13723 Edit UI - Review and Confirm step shows missing required info when there is none missing

lewischenstudio opened the pull request:
*Issue #:* /bcgov/entity#13723

*Description of changes:*
- make truthy special resolution validation flag in `parseChangeRegFiling`


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

lambert-alex reviewed (commented):
When I try this locally it does not fix the problem. Also could you add a version number please?

severinbeauvais commented on src/mixins/filing-template-mixin.ts:
```diff
@@ -866,6 +867,9 @@ export default class FilingTemplateMixin extends Mixins(DateMixin, EnumMixin) {
     this.setCorrectionStartDate(filing.changeOfRegistration.startDate || null)
 
+    // default this flag to true for change filing
+    this.setSpecialResolutionConfirmStateValidity(true)
```
How about just initializing this flag to True in state-models.ts (like the other ReviewCertify flags) ?  If there is a special reason this flag needs to be initialized as False then it should be its component's responsibility to do that.

Also, please update the app version :)

lewischenstudio replied:
I think your way is better. This flag is only used in the `CreateSpecialResolutionSummary`, which sets it to false onload, and initialize it to true in the state model works.

lewischenstudio commented:
> When I try this locally it does not fix the problem. Also could you add a version number please?

This was working on my local env, but I updated the repo with the new initialization. Can you pull and try again? App version is now added.

lambert-alex commented:
> > When I try this locally it does not fix the problem. Also could you add a version number please?
> 
> This was working on my local env, but I updated the repo with the new initialization. Can you pull and try again? App version is now added.

I tried it again and I can file now! I do get a unable to save filing error I think that might be unrelated to your fix.

![Screenshot 2022-10-04](https://user-images.githubusercontent.com/60455392/193917517-dc52c928-0692-49fe-afa0-e8579753e302.png)

lewischenstudio commented:
> > > When I try this locally it does not fix the problem. Also could you add a version number please?
> > 
> > 
> > This was working on my local env, but I updated the repo with the new initialization. Can you pull and try again? App version is now added.
> 
> I tried it again and I can file now! I do get a unable to save filing error I think that might be unrelated to your fix.
> 

This is most likely a schema error. Can you test the change filing again with another SP/GP business number? I'm not entirely sure what causes this, and I was able to file a change filing.

lambert-alex commented:
> > > > When I try this locally it does not fix the problem. Also could you add a version number please?
> > > 
> > > 
> > > This was working on my local env, but I updated the repo with the new initialization. Can you pull and try again? App version is now added.
> > 
> > 
> > I tried it again and I can file now! I do get a unable to save filing error I think that might be unrelated to your fix.
> 
> This is most likely a schema error. Can you test the change filing again with another SP/GP business number? I'm not entirely sure what causes this, and I was able to file a change filing.

Yes it is a schema error, I figured out the error I guess Province is a required field in the schema even though it is optional
